docs/DeveloperGuide.md

```diff
@@ -2,12 +2,153 @@
 
 ## Acknowledgements
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+[Original Source](https://github.com/nus-cs2113-AY2425S1/tp)
 
 ## Design & implementation
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Overall Class Diagram
+![overallClass.jpg](./Diagrams/overallClass.jpg)
+On a high level, whenever Wiagi is started, it will load SpendingList and IncomeList from Storage if it exists, else, 
+new lists would be created
+Wiagi then takes in user input via the UI class, then parse and executes the command through the parser class.
+The related output is printed through the UI class.
+At the end of the run, or when the user exits the application, Wiagi will save the lists.
+
+### Storage
+To load data from previous session:
+Within Wiagi constructor, Storage class is constructed, which will load and initialise incomes, spendings and
+password by de-serialising the text at their distinct file paths. Wiagi will then initialise it incomes and spendings
+based on the member in the Storage class.
+![storageLoad.png](./Diagrams/storageLoad.png)
+
+To save data for current session:
+After the command bye is sent by the user, incomes and spendings will be serialised and overwrite texts in
+their distinct file paths.
+![storageSave.png](./Diagrams/storageSave.png)
+
+### adding of new entry
+![addCommandSequence.jpg](./Diagrams/addCommandSequence.jpg)
+To add new entries, user will have to input the related commands.
```

Comment:
![image](https://github.com/user-attachments/assets/04f74e42-2b6e-4e96-8e9d-cd0790086ada)
The text is slightly buggy here. You might want to create a new line below the image